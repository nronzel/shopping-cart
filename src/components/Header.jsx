import React from "react";
import { Box, Button, Flex, Heading, Spacer } from "@chakra-ui/react";

function Header() {
  return (
    <Flex h="50px" w="100%" justifyContent="center" alignItems="center">
      <Flex w="80%">
        <Heading>TotallyLegitStore</Heading>
        <Spacer />
        <Button>Theme</Button>
      </Flex>
    </Flex>
  );
}

export default Header;
