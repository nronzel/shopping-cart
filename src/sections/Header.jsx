import React from "react";
import { Flex, HStack, Heading, Spacer } from "@chakra-ui/react";
import { faCartShopping } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import ThemeSwitcher from "../utils/ThemeSwitcher"

const Header = () => {
  return (
    <Flex w="100%" justifyContent="center">
      <Flex w="80%" gap={6} justifyContent="center" alignItems="center">
        <Heading cursor="default">
          <a href="/">TotallyLegitStore</a>
        </Heading>
        <Spacer />
      <HStack mr={15} spacing={5}>
        <a href="/products">Shop</a>
        <a href="/cart">
          <FontAwesomeIcon icon={faCartShopping} />
        </a>
      </HStack>
      <ThemeSwitcher />
      </Flex>
    </Flex>
  );
};

export default Header;
