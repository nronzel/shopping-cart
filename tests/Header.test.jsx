import { fireEvent, render, screen } from '@testing-library/react'
import React from "react"
import { ChakraProvider } from "@chakra-ui/react";
import Header from "../src/sections/Header.jsx"
import { userEvent } from '@testing-library/user-event';

describe("Header renders", () => {
    beforeEach(() => {
        render(<ChakraProvider><Header /></ChakraProvider>);
    })

    it("renders title", () => {
        expect(screen.getByText(/keebstore/i)).toBeInTheDocument();
    })

    it("renders the 'shop' link", () => {
        expect(screen.getByText(/shop/i)).toBeInTheDocument();
    })

    it("changes color theme", () => {
        const button = screen.getByRole("button")

        expect(document.body).toHaveClass("chakra-ui-light")

        fireEvent.click(button)

        expect(document.body).toHaveClass("chakra-ui-dark")
    })

    it("matches snapshot", () => {
        const { container } = render(<ChakraProvider><Header /></ChakraProvider>);
        expect(container).toMatchSnapshot();
    })
})
