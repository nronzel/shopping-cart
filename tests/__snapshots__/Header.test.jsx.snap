// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header renders > matches snapshot 1`] = `
<div>
  <div
    class="css-1lo8p5r"
  >
    <div
      class="css-7gyrdv"
    >
      <div
        class="css-4d8kqp"
      >
        <h2
          class="chakra-heading css-1tqj5lc"
        >
          <a
            href="/"
          >
            keebstore
          </a>
        </h2>
        <p
          class="chakra-text css-i3jkqk"
        >
          <a
            href="/products"
          >
            shop
          </a>
        </p>
        <div
          class="chakra-stack css-p73683"
        >
          <a
            href="/cart"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-cart-shopping fa-lg "
              data-icon="cart-shopping"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                fill="currentColor"
              />
            </svg>
          </a>
          <button
            class="chakra-button css-14szi6g"
            type="button"
          >
            <svg
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21.4,13.7C20.6,13.9,19.8,14,19,14c-5,0-9-4-9-9c0-0.8,0.1-1.6,0.3-2.4c0.1-0.3,0-0.7-0.3-1 c-0.3-0.3-0.6-0.4-1-0.3C4.3,2.7,1,7.1,1,12c0,6.1,4.9,11,11,11c4.9,0,9.3-3.3,10.6-8.1c0.1-0.3,0-0.7-0.3-1 C22.1,13.7,21.7,13.6,21.4,13.7z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
